src/main/scala/beam/scoring/BeamScoringFunctionFactory.scala

```diff
@@ -186,13 +186,15 @@ class BeamScoringFunctionFactory @Inject() (
           val (trip, tripIndex) = tripWithIndex
           val personId = person.getId.toString
           val tripPurpose = person.getSelectedPlan.getPlanElements.asScala
-            .filter(_.isInstanceOf[Activity])
-            .map(_.asInstanceOf[Activity])
+            .collect { case x if x.isInstanceOf[Activity] => x.asInstanceOf[Activity] }
             .lift(tripIndex + 1)
+          val tripOrigin = person.getSelectedPlan.getPlanElements.asScala
+            .collect { case x if x.isInstanceOf[Activity] => x.asInstanceOf[Activity] }
```

Comment:
It should be a little simpler, like we have here
https://github.com/LBNL-UCB-STI/beam/blob/e7a081bc1889b46a2caaa110ac3874509192ada2/src/main/scala/beam/agentsim/agents/freight/input/FreightReader.scala#L79

Reply:
Adjusted